pandas/_libs/lib.pyx

```diff
@@ -2357,3 +2354,115 @@ def fast_multiget(dict mapping, ndarray keys, default=np.nan):
             output[i] = default
 
     return maybe_convert_objects(output)
+
+
+@cython.wraparound(False)
+@cython.boundscheck(False)
+cdef inline void convert_and_set_item(object item, Py_ssize_t index,
+                                      object[:] result,
+                                      bint keep_trivial_numbers):
+    cdef:
+        bint do_convert = 1
+        float64_t float_item
+
+    if keep_trivial_numbers:
```

Comment:
is this ever not true?

Reply:
`_concat_date_cols` called twice in `make_date_converter` function, one call of these with `keep_trivial_numbers == False`